File: 프로그래머스/3/43162. 네트워크/네트워크.py
from collections import deque
def solution(n, computers):
    answer= 0
    visited=[False for _ in range(n)]
    for i in range(n):
        if(visited[i]==False):
            visited[i]=True
            q=deque([i])
            while(q):
                c_x=q.popleft()
                for n_x in range(n):
                    if(c_x==n_x):
                        continue
                    if(computers[c_x][n_x]==1 and visited[n_x]==False):
                        q.append(n_x)
                        visited[n_x]= True
            answer+=1
    return answer

File: 프로그래머스/3/43162. 네트워크/네트워크.java

class Solution {
    public void dfs(int start, int n, int[][] graph, int[] visited){
        if(visited[start] == 1) return;
        visited[start] = 1;
        for(int i = 0; i < n; i++){
            if(graph[start][i] == 1 && visited[i]== 0){
                dfs(i, n, graph, visited);
            }
        }
    }
    public int solution(int n, int[][] computers) {
        int[] visited= new int[n];
        int answer = 0;
        for(int i = 0; i < n; i++){
            if(visited[i]==0){
                dfs(i, n, computers, visited);
                answer++;
            }
        }
        return answer;
    }
}